rpc/core/net.go

```diff
@@ -18,7 +18,11 @@ func (env *Environment) NetInfo(*rpctypes.Context) (*ctypes.ResultNetInfo, error
 	env.P2PPeers.Peers().ForEach(func(peer p2p.Peer) {
 		nodeInfo, ok := peer.NodeInfo().(p2p.DefaultNodeInfo)
 		if !ok {
-			err = fmt.Errorf("peer %v has the invalid node info type: %T ", peer.ID(), peer.NodeInfo())
+			err = ErrInvalidNodeType{
```

Comment:
actually this should be OK, checked and it seems people don't rely on the error message to parse the error. We can add a note in the changelog entry for this PR (tbd) saying that if they rely on the message then this might break their client code.